### PR TITLE
Fixes #116: actions/deploy,testQuery: Refine error messages.

### LINKED ISSUE #116: [Deploy Contract] Error messaging needs to be refined
### User Story

When a user attempts to deploy a contract but rejects the transaction in Metamask, the error dialogue is overwhelming.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

image

### Current Behavior
[comment]: # (Describe what actually happened.)
See above

### Expected Behavior
[comment]: # (Describe what you expected to happen.)
Much smaller more concise message

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)
Reject a transaction in MetaMask

### Solution
Please fix in both `Quick` and `Guided` Deployers.  Issue may also exist in the `Test Query` page as well. 

## Before you `start work`
If you have ongoing work from other bounties with us where funding has not been released, please do not pick up a new issue.  We would like to involve as many contributors as possible and parallelize the work flow as much as possible.

Please also note that in order for work to be accepted, all code must be accompanied by test cases as well.

### PR DESCRIPTION
**Pull request by cryptomental**
##### Description
Errors thrown by MetaMask contain a summary and a detailed stack trace follows.
Pass only the summary to the error payload to be displayed in case an error occurs.

The change is quite safe. In case the error message consists of only one line, the full error message be shown.

##### Checklist

- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Affected core subsystem(s)
actions

##### Testing
Tested manually.

##### Refers/Fixes
Fixes: #116

### CODE REVIEW

**Comment by cryptomental** [comment]
I did an analysis of all actions and these are the two places where the large stack trace from MetaMask may be shown in the dApp error message.

**Comment by cryptomental** [comment]
One test case fails, fix in progress.

**Comment by cryptomental** [comment]
All good, @eswarasai the PR is ready for a review.

**Comment by cryptomental** [comment]
Hmm `Error: MetaMask Tx Signature: User denied transaction signature.` is clear to me: user declined the transaction and denied to sign it. I am a bit afraid that MetaMask error messages may change over time and we will have to maintain the mapping. With 1:1 error as proposed, you will always be able to track down the error easily. @pelsasser what are your thoughts?

**Comment by eswarasai** [comment]
@cryptomental -- So if you see the error in the top message notification, it is too long and also the message isn't user friendly. Even if Metamask changes the error messages, it should be fine as the default message returned from the utils function would still be the same as of today.

**Comment by cryptomental** [comment]
@eswarasai  All right, I will do as you suggested.

Please provide here a list of error mappings to implement. It can be a dictionary

```
{
  "MetaMask error message 1" : "MARKET dApp error message 1",
  "MetaMask error message 2" : "MARKET dApp error message 2",
}
```

**Comment by eswarasai** [comment]
@cryptomental -- I was thinking in the lines of having `indexOf()` check rather than entire message map. So for example, we check if the string `User denied transaction` exists in the error message and we return the custom error message in that scenario. Isn't this better?

And also I'm not aware of all the possible error scenarios that can be encountered. I think having the utils method in place should be good to close this PR. At later stages we can worry about handling it for all the known error states.

**Comment by cryptomental** [comment]
Ok, makes sense. I will tweak the PR and add the utils method over the weekend.

**Comment by cryptomental** [comment]
A helper function that returns error messages was added. As discussed it uses indexOf to return custom error messages and returns original message as a fallback. I covered the helper function by tests, screenshot also attached.

**Comment by cryptomental** [comment]
@pelsasser @eswarasai please have a look if this PR can be merged now.

**Comment by eswarasai** [comment]
@cryptomental -- Everything else looks good to me except for the part of the `utils` method name. As soon as you can update it, I should be able to merge this and close the issue. Thanks :)

File: src/util/utils.js [hunk 1]
```diff
@@ -224,3 +224,16 @@ class Order {
...
 }
 
+/**
+ * Convert MetaMask error message to dApp error message.
+ * Fallback: return original message.
+ *
+ * @param errorMessage
+ * @return dAppErrorMessage
+ */
+export const dAppErrorMessage = function (message) {
```
[the remark this answers is not in the document]
Comment (cryptomental) [reply src/util/utils.js]: ok sure!

**Comment by cryptomental** [comment]
@eswarasai I renamed to getMetamaskError and rebased the branch to current master, should be safe to merge.

**Comment by cryptomental** [comment]
I still need to update tests.

**Comment by cryptomental** [comment]
@eswarasai tests also updated. Btw. the last failure just proved that they actually work as expected! ;)

**Pull request merged**